src/test/java/team7/hrbank/unit/department/RepositoryTest.java

```diff
@@ -0,0 +1,303 @@
+package team7.hrbank.unit.department;
+
+import jakarta.persistence.EntityManager;
+import net.datafaker.Faker;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+import team7.hrbank.config.QuerydslConfig;
+import team7.hrbank.domain.department.dto.DepartmentMapperImpl;
+import team7.hrbank.domain.department.dto.DepartmentWithEmployeeCountResponseDto;
+import team7.hrbank.domain.department.dto.PageDepartmentsResponseDto;
+import team7.hrbank.domain.department.entity.Department;
+import team7.hrbank.domain.department.repository.DepartmentRepository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;
+//@DataJpaTest를 사용하는 경우 기본적으로 Auditing 기능이 활성화되지 않으므로,
+// 테스트 설정에 별도로 Auditing 설정을 포함하거나 필요한 설정 클래스를 @Import 어노테이션으로 불러오도록 구성
+
+@Import({QuerydslConfig.class, DepartmentMapperImpl.class})
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = NONE)
+@ActiveProfiles("test")
+public class RepositoryTest {
+
+    private static final Logger log = LoggerFactory.getLogger(RepositoryTest.class);
+
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    private EntityManager em;
+
+    @Autowired
+    private DepartmentMapperImpl departmentMapper;
+
+    @Test
+    @DisplayName("Fake 라이터리로 더미 데이터 생성 + 공통 단어 포함 확인")
+    void makeFakeDummy() {
+        // 원하는 엔티티 세팅 수
+        int entitySettingSize = 30;
+
+        // 공통되는 단어
+        String commonWord = "ㅎㅇ";
+
+        setting_entity_save_and_containing_name(entitySettingSize, commonWord);
+        //setting_entity_save_and_containing_description(entitySettingSize, commonWord);
+
+        List<Department> all = departmentRepository.findAll();
+
+        assertThat(all).hasSize(entitySettingSize);
+        assertThat(all).doesNotHaveDuplicates();
+        assertThat(all).allSatisfy(department -> {
+            assertThat(department.getName()).contains(commonWord);
+            //assertThat(department.getDescription()).contains(commonWord);
+        });
```

Comment:
assertThat은 메서드체이닝 지원해서 아래처럼 바꿀 수 있어요.

```suggestion
        assertThat(all)
                .hasSize(entitySettingSize)
                .doesNotHaveDuplicates()
                .allSatisfy(department -> {
                    assertThat(department.getName()).contains(commonWord);
                    //assertThat(department.getDescription()).contains(commonWord);
                });
```

Reply:
아래 코멘트에서는 하나의 테스트 당 하나만 검증하는게 좋다고 하셨는데, 이렇게 메서드체이닝을 하면 너무 많은 검증을 하는게 아닐까요? 남겨주신 2개의 코멘트가 반대되는 것 같아서 헷갈리네요

Reply:
메서드체이닝을 하더라도 하나의 목적만을 검증하기 때문에 괜찮습니다!
- findDepartments 메서드는 특정 단어가 부서명에 포함되어 있는 결과만을 반환한다.